zebra/redistribute.c

```diff
@@ -146,6 +146,34 @@ static void zebra_redistribute(struct zserv *client, int type,
 		}
 }
 
+/*
+ * Function to check if prefix is candidate for
+ * redistribute.
+ */
+static bool zebra_redistribute_check(const struct route_entry *re,
+				     struct zserv *client,
+				     const struct prefix *p,
+				     int afi)
+{
+	/*
+	 * Check if reditrubte is registered. Special check for
+	 * default route.
+	 */
+	if (re && ((is_default_prefix(p)
+	    && vrf_bitmap_check(client->redist_default[afi],
+				re->vrf_id))
+	    || vrf_bitmap_check(client->redist[afi][ZEBRA_ROUTE_ALL],
+				 re->vrf_id)
+	    || (re->instance
+	    && redist_check_instance(&client->mi_redist[afi][re->type],
+				     re->instance))
+	    || vrf_bitmap_check(client->redist[afi][re->type],
+			         re->vrf_id)))
+		return true;
```

Comment:
nit: maybe we could turn this into five `if` checks to improve readability.

Reply:
Yes -- that would be be better.